File: frontend/src/types/project.ts
export interface TechStackSpec {
  frontend?: string;
  backend?: string;
  database?: string;
  cloud?: string;
  container?: string;
}

export interface GeneratedDocumentSpec {
  id: string;
  project_id: string;
  doc_type: string;
  file_path: string;
  content: string;
  version: number;
  is_latest: boolean;
  created_at: string;
}

export interface Project {
  id: string;
  user_id: string;
  name: string;
  description?: string | null;
  tech_stack?: TechStackSpec | Record<string, string> | null;
  compliance_frameworks?: string[] | null;
  status: 'ACTIVE' | 'ARCHIVED';
  created_at: string;
  updated_at: string;
  documents?: GeneratedDocumentSpec[];
}

export interface ProjectCreatePayload {
  name: string;
  description?: string;
  tech_stack?: TechStackSpec | Record<string, string>;
  compliance_frameworks?: string[];
}

export interface ProjectUpdatePayload {
  name?: string;
  description?: string;
  tech_stack?: TechStackSpec | Record<string, string>;
  compliance_frameworks?: string[];
  status?: 'ACTIVE' | 'ARCHIVED';
}

export interface ProjectListResponse {
  projects: Project[];
  total: number;
}

File: backend/app/schemas/project.py
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict

class TechStackSpec(BaseModel):
    frontend: str = "React 18"
    backend: str = "FastAPI"
    database: str = "PostgreSQL 16"
    cloud: str = "AWS Cloud"
    container: str = "Docker + K8s"

class ProjectCreate(BaseModel):
    name: str = Field(..., min_length=2, max_length=150)
    description: str | None = None
    tech_stack: dict | TechStackSpec | None = Field(default_factory=dict)
    compliance_frameworks: list[str] = Field(default_factory=list)

class ProjectUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=150)
    description: str | None = None
    tech_stack: dict | TechStackSpec | None = None
    compliance_frameworks: list[str] | None = None
    status: str | None = None

class DocumentResponse(BaseModel):
    id: str
    project_id: str
    doc_type: str
    file_path: str
    content: str
    version: int
    is_latest: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class ProjectResponse(BaseModel):
    id: str
    user_id: str
    name: str
    description: str | None = None
    tech_stack: dict | None = None
    compliance_frameworks: list[str] | None = None
    status: str
    created_at: datetime
    updated_at: datetime
    documents: list[DocumentResponse] = []

    model_config = ConfigDict(from_attributes=True)

class ProjectListResponse(BaseModel):
    projects: list[ProjectResponse]
    total: int
